wgsl/index.bs

```diff
@@ -3615,18 +3625,18 @@ Expressions specify how values are computed.
 
 ## Type Constructor Expressions ## {#type-constructor-expr}
 
-Type constructor expressions explicitly create a value of a given [=constructible=] type.
-Type constructor expressions explicitly create a value of a given type.
+A type constructor expression explicitly creates a value of a given [=constructible=] type.
+It must match the [=syntax/constructor_expression=] grammar rule.
```

Comment:
This seems redundant and I wouldn't want a precedent of every expression saying it must obey its grammar.

Reply:
Yeah, maybe I went overboard on this.
But there truly was a confusing mix of matching "function call" vs. "constructor expression"
E.g.      vec2(1.0,2.0)  was a function call grammar rule, but is actually a type constructor.
The refactoring to use vec_prefix and mat_prefix fixed things up.

Reply:
@dneto0 couldn't constructors be treated in a unified call expression where a function call would ban type parameter not in syntax but in spec text?

Reply:
I'm not sure that's better.
primary_expression has these things that can conflict:

   ident

   type_decl paren_arg_list  // conflict because type alias and struct name are identifiers
                // so  this rule can reduce to:   ident paren_arg_list

   function_call_expression /// which is ident paren_arg_list

So yes, the grammar could drop function_call_expression and let an actual function call be covered by the type_decl paren_arg_list.  But that is really misleading.

I expect everyone who uses the grammar will work around the ambiguity, like your treesitter exception.

It all depends on who you think is using the grammar, and why.

Reply:
I've removed the line now.  It's fine for it to remain implied.